transform_and_group_strings/transform_and_group_strings.py

```diff
@@ -0,0 +1,28 @@
+def transform_and_group(names: list[str]) -> dict[int, list[str]]:
+    grouped_names = {}
+    for name in names:
+        upper_name = name.upper()
+        length = len(upper_name)
+        if length not in grouped_names:
+            grouped_names[length] = []
```

Comment:
We can drop this if we use `defaultdict`.
That reduces the number of lines and simplifies the code.